### PR TITLE
[BUGFIX] Get correct page id for a shortcut

### PR DESCRIPTION
**Pull request by KrohnMi**
When creating a shortcut in TYPO3 that points to a different root page, an incorrect link is generated.

I hope I have fixed it correctly.

### CODE REVIEW

File: Classes/Event/Listener/AfterLinkIsGeneratedListener.php [hunk 1]
```diff
@@ -31,7 +32,12 @@ public function __invoke(AfterLinkIsGeneratedEvent $event): void
             return;
         }
 
-        $pageId = $result->getLinkConfiguration()['parameter'] ?? 0;
+        $page = $result->getLinkConfiguration()['page'] ?? null;
+        if ($page != null && $page['doktype'] == PageRepository::DOKTYPE_SHORTCUT) {
```
Comment (lukaszuznanski) [inline Classes/Event/Listener/AfterLinkIsGeneratedListener.php:36]: I would love to have more strict comparision here

**Review by lukaszuznanski — CHANGES_REQUESTED** [review]
Could you cover this with tests? 
By the way: is this really an issue? I mean, it's OK to have id of shortcut page, as it's redirected by 307 HTTP to other page, or I don't understand your issue.

**Comment by KrohnMi** [comment]
@lukaszuznanski 

My use case involves two Typo3 root pages, let's call them Page A and Page B.

On Page A, i'm building a menu using the Menubuilder, and within this menu, i have links to Page B. In the Typo3 site config, i have defined the frontend URL as a.site.com and the backend URL as api.a.site.com. Similarly, Page B has frontend URL b.site.com and backend URL api.b.site.com.

When you build the menu, the links are generated as a.site.com instead of the expected b.site.com. This is happening because the Site Configuration of Page A is being used to get the frontend url, and Page B's Site Configuration is not being used as expected.

**Comment by twoldanski** [comment]
Hi @KrohnMi thanks for PR, but can you please test you issue with this PR https://github.com/TYPO3-Headless/headless/pull/675 (should solve you issue as well)

**Comment by lukaszuznanski** [comment]
@KrohnMi please check if https://github.com/TYPO3-Headless/headless/pull/675 fixes your issue

**Comment by KrohnMi** [comment]
@twoldanski No, unfortunately, your merge request could not resolve my issue.

**Comment by twoldanski** [comment]
Hi @KrohnMi could you provide example typoscript to reproduce your issue?

**Comment by lukaszuznanski** [comment]
@KrohnMi please provide some examples so we can reproduce this issue

**Pull request closed without merging**